modules/base/rendering/screenspaceinsetblackout.cpp

```diff
@@ -0,0 +1,768 @@
+        std::pair<glm::vec2, glm::vec2> rPad = calculatePadding(pointsRight);
+        std::pair<glm::vec2, glm::vec2> bPad = calculatePadding(pointsBottom);
+        std::pair<glm::vec2, glm::vec2> lPad = calculatePadding(pointsLeft);
+
+        // Adds the extra points
+        pointsTop[0] = tPad.first;
+        pointsRight[0] = rPad.first;
+        pointsBottom[0] = bPad.first;
+        pointsLeft[0] = lPad.first;
+        pointsTop.push_back(tPad.second);
+        pointsRight.push_back(rPad.second);
+        pointsBottom.push_back(bPad.second);
+        pointsLeft.push_back(lPad.second);
+
+        // Samples the spline and returns all points along the spline curve
+        std::vector<glm::vec2> splineTop = sampleSpline(pointsTop);
+        std::vector<glm::vec2> splineRight = sampleSpline(pointsRight);
+        std::vector<glm::vec2> splineBottom = sampleSpline(pointsBottom);
+        std::vector<glm::vec2> splineLeft = sampleSpline(pointsLeft);
+
+        // Translate points from range [0,1] to [-1,1] for X and Y
+        offsetCoordinates(splineTop);
+        offsetCoordinates(splineRight);
+        offsetCoordinates(splineBottom);
+        offsetCoordinates(splineLeft);
+
+        // Incoming vertex data is: top -> right -> bottom -> left (clockwise)
+        // VBO data needs to be counter-clockwise for correct winding
+        // Also adds extra point first and last for Triangle Fan drawing
+        _vboData.push_back(glm::vec2(0.0, 0.0));
+        _vboData.insert(_vboData.end(), splineTop.rbegin(), splineTop.rend());
+        _vboData.insert(_vboData.end(), splineLeft.rbegin(), splineLeft.rend());
+        _vboData.insert(_vboData.end(), splineBottom.rbegin(), splineBottom.rend());
+        _vboData.insert(_vboData.end(), splineRight.rbegin(), splineRight.rend());
+        _vboData.push_back(splineTop.back());
+    }
+}
+
+void ScreenSpaceInsetBlackout::generateTexture() {
+    // OpenGL stuff
+    glBindVertexArray(_vao);
+    glBindBuffer(GL_ARRAY_BUFFER, _vbo);
+    glBufferData(
+        GL_ARRAY_BUFFER,
+        _vboData.size() * sizeof(glm::vec2),
+        _vboData.data(),
+        GL_STATIC_DRAW
+    );
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 2, GL_FLOAT, GL_FALSE, sizeof(glm::vec2), nullptr);
+    glBindVertexArray(0);
+
+    _fboProgram->activate();
+
+    // Set uniform
+    _fboProgram->setUniform("color",
+        _blackoutShape.enableCalibrationColor ? glm::vec3(0.f, 1.f, 0.f) : glm::vec3(0.f)
+    );
+
+    // Resolution
+    GLint viewport[4];
+    glGetIntegerv(GL_VIEWPORT, viewport);
+
+    // bind FBO and Texture
+    glBindFramebuffer(GL_FRAMEBUFFER, _fbo);
+
+    bindTexture();
+
+    // Clear current buffer
+    glViewport(0, 0, 3840, 2160);
+    glClearColor(0.f, 0.f, 0.f, 0.f);
+    glClear(GL_COLOR_BUFFER_BIT);
+
+    // Draw
+    glBindVertexArray(_vao);
+    glDrawArrays(GL_TRIANGLE_FAN, 0, static_cast<gl::GLsizei>(_vboData.size()));
+    glBindVertexArray(0);
```

Comment:
Some redundant comments here
```suggestion
    _fboProgram->setUniform("color",
        _blackoutShape.enableCalibrationColor ? glm::vec3(0.f, 1.f, 0.f) : glm::vec3(0.f)
    );

    GLint viewport[4];
    glGetIntegerv(GL_VIEWPORT, viewport);

    glBindFramebuffer(GL_FRAMEBUFFER, _fbo);

    bindTexture();

    // Clear current buffer
    glViewport(0, 0, 3840, 2160);
    glClearColor(0.f, 0.f, 0.f, 0.f);
    glClear(GL_COLOR_BUFFER_BIT);

    glBindVertexArray(_vao);
    glDrawArrays(GL_TRIANGLE_FAN, 0, static_cast<gl::GLsizei>(_vboData.size()));
    glBindVertexArray(0);
```